nixos/modules/services/networking/syncthing.nix

```diff
@@ -144,8 +144,8 @@ let
   ''
   +
   (lib.optionalString (cfg.guiPasswordFile != null) ''
-     pw_bcrypt=$(${pkgs.mkpasswd}/bin/mkpasswd -m bcrypt --stdin <"${cfg.guiPasswordFile}")
-     curl -X PATCH --variable '%pw_bcrypt' --expand-json '{ "password": "{{pw_bcrypt}}" }' ${curlAddressArgs "/rest/config/gui"}
+     ${pkgs.mkpasswd}/bin/mkpasswd -m bcrypt --stdin <"${cfg.guiPasswordFile}" | tr -d "\n" >$RUNTIME_DIRECTORY/password_bcrypt
```

Comment:
I think if mkpasswd reads the trailing newline from guiPasswordFile, removing the newline must be done before passing the data to mkpasswd, not after? Or maybe both should be stripped?

Reply:
I checked the implementation of `mkpasswd` and it only reads the first line of the input (file) _excluding_ the line terminator. (So not stripping the newline before passing the input to mkpasswd is fine.)